fluent-bundle/src/bundle.ts

```diff
@@ -1,12 +1,13 @@
 import { resolveComplexPattern } from "./resolver.js";
 import { Scope } from "./scope.js";
 import { FluentResource } from "./resource.js";
-import { FluentNone, FluentType, FluentFunction } from "./types.js";
+import { FluentValue, FluentNone, FluentFunction } from "./types.js";
 import { Message, Term, Pattern } from "./ast.js";
+import { NUMBER, DATETIME } from "./builtins.js";
 
 export type TextTransform = (text: string) => string;
 
-export type FluentArgument = string | number | Date | FluentType;
+export type FluentArgument = string | number | Date | FluentValue;
```

Comment:
Do you want to keep doubling the `string` here? Or should we do a `FluentType<any>` here?

Reply:
It's redundant because of the implementation detail, but I kept it because semantically I wanted to say: an argument can either be:

  - an instance of `FluentValue`, or
  - a JS primitive or a `Date`, in which case we'll take care of converting it to a proper `FluentValue`.

Do you think it's worth keeping this distinction in code?

Reply:
The semantics of `FluentType<any>` could be:

A JS primitive or a `Date`, or a custom FluentType, or one with options.

I don't mind either way, it's just that it might say just slightly different things?

Reply:
What if I expressed it as follows? Does it communicate the semantics better?

```ts
type NativeArgument = string | number | Date;
export type FluentArgument = FluentValue | NativeArgument;
```

Reply:
Yeah, that's good.

Reply:
Thanks!